src/picongpu/include/plugins/hdf5/writer/Field.hpp

```diff
@@ -135,14 +155,80 @@ struct Field
                                                DomainCollector::GridType,
                                                tmpArray);
 
-            /*simulation attributes for data*/
-            ColTypeDouble ctDouble;
+            /* attributes */
+            params->dataCollector->writeAttribute(params->currentStep,
+                                                  splashFloatXType, datasetName.str().c_str(),
+                                                  "position",
+                                                  1u, Dimensions(simDim,0,0),
+                                                  &(*inCellPosition.at(n).begin()));
 
             params->dataCollector->writeAttribute(params->currentStep,
                                                   ctDouble, datasetName.str().c_str(),
-                                                  "sim_unit", &(unit.at(n)));
+                                                  "unitSI", &(unit.at(n)));
         }
         __deleteArray(tmpArray);
+
+
+        params->dataCollector->writeAttribute(params->currentStep,
+                                              ctDouble, recordName.c_str(),
+                                              "unitDimension",
+                                              1u, Dimensions(7,0,0),
+                                              &(*unitDimension.begin()));
+
+        params->dataCollector->writeAttribute(params->currentStep,
+                                              splashFloatXType, recordName.c_str(),
+                                              "timeOffset", &timeOffset);
+
+        std::string geometry("cartesian");
+        ColTypeString ctGeometry(geometry.length());
+        params->dataCollector->writeAttribute(params->currentStep,
+                                              ctGeometry, recordName.c_str(),
+                                              "geometry", geometry.c_str());
+
+        std::string dataOrder("C");
+        ColTypeString ctDataOrder(dataOrder.length());
+        params->dataCollector->writeAttribute(params->currentStep,
+                                              ctDataOrder, recordName.c_str(),
+                                              "dataOrder", dataOrder.c_str());
+
+        typedef char MyChar2[2];
+        MyChar2 *axisLabels = new MyChar2[simDim];
+        ColTypeString ctAxisLabels(1);
+        for( uint32_t d = 0; d < simDim; ++d )
+        {
+            /* \todo is the order correct? */
+            axisLabels[d][0] = char(120 + d); // x, y, z
```

Comment:
ok, `char( 'x' + d )` is fully valid and compiles with `-Wall -Wextra -pedantic` in `g++ 4.9.2`:

```c++
#include <iostream>

int main()
{
  char c = 'a';
  unsigned int d = 1;

  c = char('x' + d);
  std::cout << c << std::endl;
  return 0;
}
```